src/reynir/bintokenizer.py

```diff
@@ -1342,6 +1342,9 @@ def is_interesting(token) -> bool:
     def can_concat(token) -> bool:
         """ Return True if the token content can be concatenated onto
             an existing entity name """
+        # Non-capitalized function words that can appear within entity names
+        if token.txt in ["in", "a", "an", "for", "and", "the", "for", "on", "of"] or token.txt in FOREIGN_MIDDLE_NAME_SET: 
```

Comment:
Convert list to a frozenset() constant for efficiency

Reply:
Done, added ENTITY_MIDDLE_NAME_SET